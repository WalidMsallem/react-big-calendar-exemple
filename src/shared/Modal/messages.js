/*
 * Modal Messages
 *
 * This contains all the text for the Modal component.
 */

import { defineMessages } from 'react-intl'

export const scope = 'app.components.Modal'

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'This is the Modal component!',
  },
  cancel: {
    id: `${scope}.cancel`,
    defaultMessage: 'Cancel',
  },
  submit: {
    id: `${scope}.submit`,
    defaultMessage: 'Submit',
  },
  edit: {
    id: `${scope}.edit`,
    defaultMessage: 'Edit',
  },
})
