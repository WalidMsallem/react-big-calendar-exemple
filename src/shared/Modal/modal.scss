    .ant-modal-content {

        .ant-modal-close {
            position: absolute;
            top: -16px !important;
            right: 16px !important;

            .ant-modal-close-x {
                width: 0 !important;
                height: 0 !important;
                font-size: 15px !important;
                line-height: 0 !important;

                .icon-container {
                    height: 32px;
                    width: 32px;
                    border-radius: 50%;
                    background-color: #170ff0;
                    justify-content: center;
                    align-items: center;
                    display: flex;
                    color: #fff;
                }
            }
        }

        .ant-modal-header {
            padding: 36px 24px !important;

            .ant-modal-title {
                font-family: Roboto !important;
                font-size: 1.625rem !important;
                color: #202224 !important;
                text-align: center;
                line-height: 0 !important;
            }
        }

        .ant-modal-footer {
            padding: 0 25px 15px !important;
            border: 0 !important;

            .btn {
                font-family: Roboto !important;
                border-radius: 3px;
                font-size: 0.875rem;
                min-height: 45px;
            }

            .cancel-btn {
                border-radius: 3px;
                background-color: #ffffff;
                color: #170ff0;
                min-width: 90px;

            }

            .submit-btn {
                border: solid 1px #170ff0;
                background-color: #170ff0;
                color: #fefefe;
                min-width: 115px;

            }

            .ant-btn:hover,
            .ant-btn:focus {
                border-color: #d9d9d9 !important;
            }
        }

        .ant-modal-body {
            padding: 10px 25px 10px 25px !important;
          
        }


        .ant-form-item label {
            font-size: 0.8125rem !important;
            color: #979797 !important;
        }

        .ant-form-item-required::before {
            display: none !important;
        }


        // <InputNumber />

        .ant-input-number {
            width: 100% !important;
            height: 46px !important;
            padding: 4px 11px;
            color: #393c40 !important;
            font-size: 0.875rem !important;
            line-height: 0 !important;
            background-color: #fff !important;
            border-radius: 3px;
            display: flex;
            align-items: center;

            .ant-input-number-input-wrap {
                width: 100%;
            }

            &.x-required-field {
                border-color: #c7163c !important;
            }
        }

        // <Input />

        .ant-input {
            width: 100% !important;
            height: 46px !important;
            padding: 4px 11px;
            color: #393c40 !important;
            font-size: 0.875rem !important;
            line-height: 0 !important;
            background-color: #fff !important;
            border-radius: 3px;
            display: flex;
            align-items: center;

            .ant-input-input-wrap {
                width: 100%;
            }

            &.x-required-field {
                border-color: #c7163c !important;
            }
        }


        // <Select />

        .ant-select-selection {
            width: 100% !important;
            height: 46px !important;
            padding: 4px 11px;
            color: #393c40 !important;
            font-size: 0.875rem !important;
            line-height: 0 !important;
            background-color: #fff !important;
            border-radius: 3px;
            display: flex;
            align-items: center;

            .ant-select-selection-input-wrap {
                width: 100%;
            }

            &.x-required-field {
                border-color: #c7163c !important;
            }
        }
    }

    .ant-modal {
        top :unset!important;
        display: flex !important;
        flex-direction: column!important;
        justify-content: center!important;
        padding-bottom: unset!important;
        height: 100vh;
    }