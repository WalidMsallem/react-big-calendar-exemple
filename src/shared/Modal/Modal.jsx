/**
 *
 * Modal
 *
 */

import React from 'react'
import { Modal as ModalAntd, Button, Icon } from 'antd'
import './modal.scss'

function Modal({
  title,
  footer,
  visible,
  handleOk,
  handleCancel,
  loading,
  submitButtonTitle,
  children,
  ...props
}) {
  return (
    <ModalAntd
      title={title}
      visible={visible}
      onOk={handleOk}
      onCancel={handleCancel}
      footer={
        footer || [
          <Button key="1" className="btn cancel-btn" onClick={handleCancel}>
           Cancel
          </Button>,
          <Button
            key="2"
            className="btn submit-btn"
            loading={loading}
            onClick={handleOk}
          >
            {submitButtonTitle}
          </Button>,
        ]
      }
      closeIcon={[
        <div key="3" className="icon-container">
          <Icon type="close" className="icon-container-close" />
        </div>,
      ]}
      {...props}
    >
      {children}
    </ModalAntd>
  )
}


Modal.defaultProps = {
  submitButtonTitle: 'Submit',
  loading: false,
}

export default Modal
