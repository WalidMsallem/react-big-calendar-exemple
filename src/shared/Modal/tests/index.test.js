import React, { Fragment } from 'react'
import { render } from 'react-testing-library'
import { Provider } from 'react-redux'
import { browserHistory, BrowserRouter } from 'react-router-dom'
import { IntlProvider } from 'react-intl'
import Modal from '../index'
import { DEFAULT_LOCALE } from '../../../../i18n'
import configureStore from '../../../../configureStore'
import 'jest-prop-type-error'

const initialProps = {
  title: <Fragment />,
  footer: <Fragment />,
  visible: false,
  handleOk: jest.fn(),
  handleCancel: jest.fn(),
  loading: false,
  submitButtonTitle: <Fragment />,
  children: {},
}

describe('<Modal />', () => {
  let store
  beforeAll(() => {
    store = configureStore({}, browserHistory)
  })

  it('Expect to crash if propTypes failed', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <IntlProvider locale={DEFAULT_LOCALE}>
            <Modal {...initialProps} />
          </IntlProvider>
        </BrowserRouter>
      </Provider>,
    )
  })

  it('Expect to not log in console', () => {
    const spy = jest.spyOn(global.console, 'log')
    render(
      <Provider store={store}>
        <BrowserRouter>
          <IntlProvider locale={DEFAULT_LOCALE}>
            <Modal {...initialProps} />
          </IntlProvider>
        </BrowserRouter>
      </Provider>,
    )
    expect(spy).not.toHaveBeenCalled()
  })

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error')
    render(
      <Provider store={store}>
        <BrowserRouter>
          <IntlProvider locale={DEFAULT_LOCALE}>
            <Modal {...initialProps} />
          </IntlProvider>
        </BrowserRouter>
      </Provider>,
    )
    expect(spy).not.toHaveBeenCalled()
  })

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <BrowserRouter>
          <IntlProvider locale={DEFAULT_LOCALE}>
            <Modal {...initialProps} />
          </IntlProvider>
        </BrowserRouter>
      </Provider>,
    )
    expect(firstChild).toMatchSnapshot()
  })
})
