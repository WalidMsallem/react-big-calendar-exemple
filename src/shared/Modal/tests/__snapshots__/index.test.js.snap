// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> Should render and match the snapshot 1`] = `null`;
